app/controllers/concerns/workflow_execution_actions.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+# Common workflow execution actions
+module WorkflowExecutionActions
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_default_tab, only: :show
+    before_action :current_page, only: :index
+    before_action :workflow_execution, only: %i[show cancel destroy]
+  end
+
+  TABS = %w[summary params files].freeze
+
+  def index
+    authorize! @namespace, to: :view_workflow_executions? unless @namespace.nil?
+
+    @q = load_workflows.ransack(params[:q])
+    set_default_sort
+    @pagy, @workflows = pagy_with_metadata_sort(@q.result)
+  end
+
+  def show
+    authorize! @namespace, to: :view_workflow_executions? unless @namespace.nil?
+
+    if @tab == 'files'
+      @samples_worfklow_executions = @workflow_execution.samples_workflow_executions
+      @attachments = Attachment.where(attachable: @workflow_execution)
+                               .or(Attachment.where(attachable: @samples_worfklow_executions))
+    elsif @tab == 'params'
+      @workflow = Irida::Pipelines.find_pipeline_by(@workflow_execution.metadata['workflow_name'],
+                                                    @workflow_execution.metadata['workflow_version'])
+    end
+  end
+
+  def destroy # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    WorkflowExecutions::DestroyService.new(@workflow_execution, current_user).execute
+
+    if @workflow_execution.deleted? && params[:redirect]
+      flash[:success] = t('.success', workflow_name: @workflow_execution.metadata['workflow_name'])
+      redirect_to workflow_executions_path(format: :html)
```

Comment:
**nitpick (blocking):** Please add a `redirect_path` method so that from the projects workflow executions we redirect to project workflow executions instead of workflow executions.

Reply:
Updated the controllers using this concern to contain the redirect_path in 3db41ad0245de038956cfba245d7855ce5e5ba9c